obspy/signal/tests/test_filter.py

```diff
@@ -239,6 +241,30 @@ def test_lowpass_cheby_2(self):
         # be 0 (1dB ripple) before filter ramp
         self.assertGreater(h_db[freq < 25].min(), -1)
 
+    def test_bandpass_high_corner_at_nyquist(self):
+        """
+        Check that using exactly Nyquist for high corner gives correct results.
+        See #1451.
+        """
+        tr = read()[0]
+        data = tr.data[:1000]
+
+        df = tr.stats.sampling_rate
+        nyquist = df / 2.0
+
+        for low_corner in (6.0, 8.55, 8.59):
+            for corners in (3, 4, 5, 6):
+                expected = bandpass(data, low_corner, nyquist - 1e-5,
+                                    df=df, corners=corners)
+                got1 = bandpass(data, low_corner, nyquist,
+                                df=df, corners=corners)
+                with warnings.catch_warnings(record=True):
+                    got2 = bandpass(data, low_corner, nyquist + 1.78,
+                                    df=df, corners=corners)
```

Comment:
I think it would make sense to also test the warning message just to make sure it does not run into a numpy or other warning.